README.md

```diff
@@ -399,6 +399,39 @@ app.get('/auth', passport.authenticate('oidc', [options]));
 app.get('/auth/cb', passport.authenticate('oidc', { successRedirect: '/', failureRedirect: '/login' }));
 ```
 
+### Passport with PKCE
+The Client instance must have the property `token_endpoint_auth_method` setted to `none`.
+See [Proof Key for Code Exchange by OAuth Public Clients][feature-pkce] spec.
+
+
+```js
+const client = new issuer.Client({
+  client_id: 'zELcpfANLqY7Oqas',
+  token_endpoint_auth_method: 'none'
+});
+
+const Strategy = require('openid-client').Strategy;
+
+passport.use('oidc', new Strategy({ client, [params], [passReqToCallback] }, (tokenset, userinfo, done) => {
+    //... same as the previous example
+}));
+
+const verifier = 'foo'; // random PKCE verifier
+
+// start authentication request
+app.get('/auth', passport.authenticate('oidc', {
+  code_challenge: yourChallengeFunction(verifier),
+  code_challenge_method: 'S256' // or 'plain'
+}));
+
+// authentication callback
+app.get('/auth/cb', passport.authenticate('oidc', {
+    code_verifier: verifier,
+    successRedirect: '/',
+    failureRedirect: '/login'
+}));
+```
+
```

Comment:
I feel like the interaction could be made much more simple for the developer. Few options

1) starting the authorization could be made with a verifier. The client can calculate the S256 challenge on its own. There's no need for `yourChallengeFunction` since its standardized, isnt it?
2) PKCE could be automatically used if request is started with code_challenge_method (or simply by setting a strategy instance e.g. `option.usePKCE = true // or "plain", "S256". true would try and look for S256 in issuer configuration, if not found, would use plain, if not exposed, throw?`, verifier would be generated by the strategy and stored in the session just like nonce and states are.